contracts/extensions/VotingReputation.sol

```diff
@@ -825,14 +831,19 @@ contract VotingReputation is DSMath, PatriciaTreeProofs {
   }
 
   function hashExpenditureSlot(bytes memory action) internal returns (bytes32 hash) {
-    assert(getSig(action) == CHANGE_FUNCTION);
+    bytes32 value;
 
     assembly {
-      // Hash all but last (value) bytes32
+      // Hash all but the domain proof and value
       // Recall: mload(action) gives length of bytes array
       // So skip past the three bytes32 (length + domain proof),
-      //   and the last bytes32 (value), plus 4 bytes for the sig.
-      hash := keccak256(add(action, 0x64), sub(mload(action), 0x64))
+      //   plus 4 bytes for the sig. Subtract the same from the end, less
+      //   the length bytes32. The value itself is located at 0xe4,
+      //   so zero that out and restore it after hashing.
+      value := mload(add(action, 0xe4))
+      mstore(add(action, 0xe4), 0x0)
+      hash := keccak256(add(action, 0x64), sub(mload(action), 0x44))
+      mstore(add(action, 0xe4), value)
```

Comment:
1. Do we actually need to restore it? The action as a whole has been copied in to memory here, right, and the original action is still safe in storage?
2. Why are we now skipping the last word of the action? Is that a leftover from trying out other things?

Reply:
1. Good point, let's try leaving it out.

2. I think this actually hashes the action correctly. If `mload(action)` gives us the length of the array _excluding_ the first bytes32, then `add(action, 0x64)` puts us right after the domain proof, while `sub(mload(action), 0x44)` gives us the length of the array less the domain proof. The difference is that `action` points us to the start of the action _including_ the length bytes32, while `mload(action)` gives us the length _excluding_ the length bytes32. Am I mistaken?

```
0x[ length ][ sig ][ domain proof ][ data ]
```

Reply:
Ah, I see, yeah, you're right.

Reply:
Yup, leaving out the value works great.